---
title: "Litter Size Evolution"
output:
  pdf_document: default
  github_document: default
---
Data: 1006 out of 2278 species (all species within Rodentia that I have data for)
Plotted: 819 species are phylogenetically resolved, thus plotted here 
Consensus tree extracted from an Upham et al. paper (Jetz lab)  

This version includes all families and all species that I can find. I labeled suprafamilies and a few common families (Acomys, Mus, Rattus). 

The labeling are based on [this paper](https://bmcevolbiol.biomedcentral.com/articles/10.1186/1471-2148-8-321). There is a minor taxonomy conflict, the relationship between Heteromyidae and Geomyidae outlined by the paper is different from this tree I used. I am leaning towards the tree, as it was published in 2019, time calibrated and used 31 genes instead of 2 to construct.


Method: REML, metadata attached at the end. The confidence interval is quite wide. One of the Brownian motion parameter is Nah, and I am still trying to figure out what that means. 


Thoughts:
The earlier ancestors all have average litter size of 2.88 or 2.89, and then evolve either into higher or smaller litter size. There are a few outliers, such as Heteroephalus_glaber, which has a litter size of 11.3, among species with litter size of 3. I am not sure how this reconstruction can disqualify the hypothesis that smaller litter size is the driving force. Would love to discuss.  


```{r setup, echo = FALSE, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ape)
library(ggtree)
library(knitr)
library(phytools)
library(tidyverse)
library(magrittr)
```

```{r fig.width=15, fig.height = 200, include = FALSE} 
  tree_global = read.tree('RAxML_bipartitions.result_FIN4_raw_rooted_wBoots_4098mam1out_OK.newick')
  n_tips = length(tree_global$tip.label)
  root_node = tree_global$edge[(!tree_global$edge[,1] %in% tree_global$edge[,2]),1] %>% unique()
  root_age = max( dist.nodes(tree_global)[,root_node] )
  
  label_offset = 0.02 * root_age
  tree_width = 5 * root_age
```

```{r, include = FALSE}
## drop all the non-relevant tips

charstate = read.csv('1128_Completedata.csv') 
charstate
charstate = charstate%>%mutate(Average = as.numeric(Average))

tips <- tree_global$tip.label
tips_short = str_replace(tips, "[A-Z_]+$", "")
tree_global$tip.label = tips_short

tip_names = data.frame(Name_long=tips, Name=tips_short)
charstate %<>% left_join(tip_names)

missing_in_charstate = tree_global$tip.label[! tree_global$tip.label %in% charstate$Name]
 
tree_global = ape::drop.tip(tree_global, missing_in_charstate)

charstate %<>% filter(!is.na(Name_long))

rank <- match( charstate[,1], tree_global$tip.label )

charstate = charstate[match( tree_global$tip.label, charstate[,1] ), ] #the data order matches, checked manually as well.

 if(! all( charstate[,1] ==tree_global$tip.label ) ){
      stop("Error: tip labels and dataframe rows not in same order!")
    }

```

```{r,  fig.height = 250, fig.width = 50, echo = FALSE}

litter_ancestral = 
  ace(
  charstate$Average, 
  tree_global,
  type = "continuous",
  method = "REML",
  model = "BM"
)

node_values = c(charstate$Average, round(litter_ancestral$ace, digits= 2)) # datastructure for plotting the tips

#for clade

MRCA_dipodidae = getMRCA(tree_global, c("Zapus_princeps", "Sicista_tianshanica"))
MRCA_muridae = getMRCA(tree_global, c("Spalax_microphthalmus","Melanomys_caliginosus"))
MRCA_anomaluridae = getMRCA(tree_global, c("Anomalurus_beecrofti", "Idiurus_macrotis"))
MRCA_pedetidae = getMRCA(tree_global, c("Pedetes_capensis", "Pedetes_surdaster"))

MRCA_geomyidae = getMRCA(tree_global, c("Geomys_texensis", "Thomomys_talpoides"))
MRCA_heteromyidae = getMRCA(tree_global, c("Heteromys_oresterus", "Dipodomys_phillipsii"))
MRCA_hystricognathi = getMRCA(tree_global, c("Trichys_fasciculata", "Cavia_tschudii"))
MRCA_castoridae = getMRCA(tree_global, c("Castor_canadensis", "Castor_fiber"))

MRCA_ctenodactylidae = getMRCA(tree_global, c("Ctenodactylus_gundi", "Massoutiera_mzabi"))
MRCA_sciuridae = getMRCA(tree_global, c("Tamiasciurus_mearnsi", "Ratufa_bicolor"))
MRCA_aplodontidae = getMRCA(tree_global, c("Aplodontia_rufa", "Aplodontia_rufa"))
MRCA_gliridae = getMRCA(tree_global, c("Glirulus_japonicus", "Dryomys_nitedula"))

#for highlight
MRCA_acomys = getMRCA(tree_global, c("Acomys_cahirinus", "Acomys_spinosissimus"))
MRCA_mus = getMRCA(tree_global, c("Mus_pahari", "Mus_tenellus"))
MRCA_rattus = getMRCA(tree_global, c("Rattus_exulans", "Rattus_praetor"))


globaltree = ggtree(tree_global, layout = 'circular' ) + 
  geom_tiplab( fontface = "italic", size= 1.5, offset = 0.005) + 
  xlim(0, 1.2) + 
  geom_text2(aes(label = node_values), nudge_x = 0.005, size = 1.5) +
  geom_tippoint(aes(color=node_values), size= 1.5, alpha= 0.8 ) + 
  geom_nodepoint( aes(color=node_values), size= 1.5, alpha= 0.8 ) + 
  scale_color_gradientn(colours = rainbow(3)) +
  theme(legend.text=element_text(size= 40),
        legend.title=element_text(size=60),
        legend.key.size= unit(3,"cm"), 
        legend.position= "top") + 
  geom_cladelabel(node=MRCA_muridae, label="Muridae", offset = 0.05, fontsize = 8, color = "blue", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_dipodidae, label="Dipodidae", offset = 0.1, fontsize = 8, color = "blue",barsize = 1.5) + 
  geom_cladelabel(node=MRCA_anomaluridae, label="Anomaluridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5 ) +
  geom_cladelabel(node=MRCA_pedetidae, label="Pedetidae", offset = 0.12, fontsize = 8, color = "blue", barsize = 1.5 ) +
  
  geom_cladelabel(node=MRCA_heteromyidae, label="Heteromyidae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5 ) +
  geom_cladelabel(node=MRCA_geomyidae, label="Geomyidae", offset = 0.05, fontsize = 8, color = "blue", barsize = 1.5 ) +
  geom_cladelabel(node=MRCA_hystricognathi, label="Hystricognathi", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_sciuridae, label="Sciuridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_gliridae, label="Gliridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_castoridae, label="Castoridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_ctenodactylidae, label="Ctenodactylidae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + 
  
  geom_cladelabel(node=MRCA_acomys, label="Acomys", offset = 0.02, fontsize = 8, color = "red", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_mus, label="Mus", offset = 0.02, fontsize = 8, color = "red", barsize = 1.5) + 
  geom_cladelabel(node=MRCA_rattus, label="Rattus", offset = 0.02, fontsize = 8, color = "red", barsize = 1.5) 

globaltree

```
Metadata about this reconstruction: 

- `$sigma2` is the estimate of the Brownian Motion parameter.
- `$CI95` are the confidence intervals on the ancestral character state reconstructions.

Residual log-likelihood: `r litter_ancestral$resloglik` 

Sigma2: `r litter_ancestral$sigma2`


CI95: 
`r kable(litter_ancestral$CI95, col.names = c('lower bound', 'upper bound'))`



<!-- BELOW PIC CODE, not the method preferred-->
<!-- Ancestral Reconstruction using Phylogenetic Independent Contrast -->

<!-- ```{r,  fig.height = 250, fig.width = 50, echo = FALSE} -->
<!-- ##### The following are for pic method, not sure to include or not -->

<!-- litter_ancestral_pic = -->
<!--   ace( -->
<!--   charstate$Average, -->
<!--   tree_global, -->
<!--   type = "continuous", -->
<!--   method = "pic", -->
<!--   model = "BM" -->
<!-- ) -->

<!-- node_values_pic = c(charstate$Average, round(litter_ancestral_pic$ace, digits= 2)) -->

<!-- globaltree_pic = ggtree(tree_global, layout = 'circular' ) + -->
<!--   geom_tiplab( fontface = "italic", size= 1.5, offset = 0.01) + -->
<!--   xlim(0, 1.2) + -->
<!--   geom_text2(aes(label = node_values_pic), nudge_x = 0.01, size = 1.5) + -->
<!--   geom_tippoint(aes(color=node_values_pic), size= 1.5, alpha= 0.8 ) + -->
<!--   geom_nodepoint( aes(color=node_values_pic), size= 1.5, alpha= 0.8 ) + -->
<!--   scale_color_gradientn(colours = rainbow(3)) + -->
<!--   theme(legend.text=element_text(size= 40), -->
<!--         legend.title=element_text(size=60), -->
<!--         legend.key.size= unit(3,"cm"), -->
<!--         legend.position= "top") + -->
<!--   geom_cladelabel(node=MRCA_muridae, label="Muridae", offset = 0.05, fontsize = 8, color = "blue", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_dipodidae, label="Dipodidae", offset = 0.1, fontsize = 8, color = "blue",barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_anomaluridae, label="Anomaluridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5 ) + -->
<!--   geom_cladelabel(node=MRCA_pedetidae, label="Pedetidae", offset = 0.12, fontsize = 8, color = "blue", barsize = 1.5 ) + -->

<!--   geom_cladelabel(node=MRCA_heteromyidae, label="Heteromyidae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5 ) + -->
<!--   geom_cladelabel(node=MRCA_hystricognathi, label="Hystricognathi", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_sciuridae, label="Sciuridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_gliridae, label="Gliridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_castoridae, label="Castoridae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_ctenodactylidae, label="Ctenodactylidae", offset = 0.1, fontsize = 8, color = "blue", barsize = 1.5) + -->

<!--   geom_cladelabel(node=MRCA_acomys, label="Acomys", offset = 0.02, fontsize = 8, color = "red", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_mus, label="Mus", offset = 0.02, fontsize = 8, color = "red", barsize = 1.5) + -->
<!--   geom_cladelabel(node=MRCA_rattus, label="Rattus", offset = 0.02, fontsize = 8, color = "red", barsize = 1.5) -->


<!-- globaltree_pic -->

<!-- ``` -->
<!-- Metadata about this reconstruction: -->

<!-- Residual log-likelihood: *litter_ancestral_pic$resloglik* -->
<!-- Sigma2: *litter_ancestral_pic$sigma2* -->
<!-- CI95: *litter_ancestral_pic$CI95* -->

<!-- - `$sigma2` is the estimate of the Brownian Motion parameter. -->
<!-- - `$CI95` are the confidence intervals on the ancestral character state reconstructions. -->

